### PR TITLE
Update from fontconfig 2.12 to 2.13

### PR DESCRIPTION
**Pull request by NHDaly**
Update Julia Homebrew.jl's `fontconfig` version to the latest version: `2.13.0`.

Cairo.jl is currently failing to build on mac (failure: https://travis-ci.org/JuliaGraphics/Cairo.jl/jobs/415954409) because it requires `fontconfig >= 2.13`. This homebrew Formula is forcing `2.12` to be installed instead, so pangocairo fails to build.

For more information, please see this thread:
https://github.com/JuliaGraphics/Cairo.jl/issues/230

In this PR, i just blindly bumped the `url` and `sha` to what I see in my `homebrew-core/Formula/fontconfig.rb`. But maybe there are other lines that need to change as well? Here are the complete contents of my `~/.julia/v0.6/Homebrew/deps/usr/Library/Taps/homebrew/homebrew-core/Formula/fontconfig.rb`, which i got via ``julia0.6> Homebrew.brew(`edit fontconfig`)``:

```ruby
class Fontconfig < Formula
  desc "XML-based font configuration API for X Windows"
  homepage "https://wiki.freedesktop.org/www/Software/fontconfig/"
  url "https://www.freedesktop.org/software/fontconfig/release/fontconfig-2.13.0.tar.bz2"
  sha256 "91dde8492155b7f34bb95079e79be92f1df353fcc682c19be90762fd3e12eeb9"

  # The bottle tooling is too lenient and thinks fontconfig
  # is relocatable, but it has hardcoded paths in the executables.
  bottle do
    sha256 "66ad096a51d6253be02b4d3df0299d422c8bd1bccf571ed64a8b8b21a2e77bc7" => :high_sierra
    sha256 "d822bc26c6b556606087e5807293492814fedf9d408c857006b0886608010400" => :sierra
    sha256 "a8016f2aff75677bab388a0ab138b4528d787c44c337f6ee7236e0de4a1cb268" => :el_capitan
  end

  pour_bottle? do
    reason "The bottle needs to be installed into /usr/local."
    # c.f. the identical hack in lua
    # https://github.com/Homebrew/homebrew/issues/47173
    satisfy { HOMEBREW_PREFIX.to_s == "/usr/local" }
  end

  head do
    url "https://anongit.freedesktop.org/git/fontconfig", :using => :git

    depends_on "autoconf" => :build
    depends_on "automake" => :build
    depends_on "libtool" => :build
  end

  depends_on "pkg-config" => :build
  depends_on "freetype"

  def install
    # Remove for > 2.13.0
    # Upstream issue from 6 Mar 2018 "2.13.0 erroneously requires libuuid on macOS"
    # See https://bugs.freedesktop.org/show_bug.cgi?id=105366
    ENV["UUID_CFLAGS"] = " "
    ENV["UUID_LIBS"] = " "

    # Remove for > 2.13.0
    # Same effect as upstream commit from 10 Mar 2018 "Add uuid to
    # Requires.private in .pc only when pkgconfig macro found it"
    inreplace "configure",
      'PKGCONFIG_REQUIRES_PRIVATELY="$PKGCONFIG_REQUIRES_PRIVATELY uuid"', ""

    font_dirs = %w[
      /System/Library/Fonts
      /Library/Fonts
      ~/Library/Fonts
    ]

    if MacOS.version >= :sierra
      font_dirs << Dir["/System/Library/Assets/com_apple_MobileAsset_Font*"].max
    end

    system "autoreconf", "-iv" if build.head?
    system "./configure", "--disable-dependency-tracking",
                          "--disable-silent-rules",
                          "--enable-static",
                          "--with-add-fonts=#{font_dirs.join(",")}",
                          "--prefix=#{prefix}",
                          "--localstatedir=#{var}",
                          "--sysconfdir=#{etc}"
    system "make", "install", "RUN_FC_CACHE_TEST=false"
  end

  def post_install
    ohai "Regenerating font cache, this may take a while"
    system "#{bin}/fc-cache", "-frv"
  end

  test do
    system "#{bin}/fc-list"
  end
end
```

-------------------

And actually, that said, is there any need for this separate file? Can we just use the default Homebrew-installed Ruby file?

--------------------------------

When opening an issue, please ping `@staticfloat`.

He does not receive emails automatically when new issues are created.
Without this ping, your issue may slip through the cracks!
If no response is garnered within a week, feel free to ping again.

### CODE REVIEW

**Comment by NHDaly** [comment]
Pinging you, @staticfloat! :) Lemme know if I've done this right..

**Comment by staticfloat** [comment]
Yes, you have.  Can you please change the `sha256` lines within the `bottle do` stanza to match your homebrew installation's formula?  I'm just going to rip off their bottles.

**Comment by NHDaly** [comment]
> Can you please change the sha256 lines within the bottle do stanza to match your homebrew installation's formula
Happily. :) Done!

**Comment by NHDaly** [comment]
And sorry, this question might've gotten lost in the above big message. I'm posting it again. 😸 

> And actually, that said, is there any need for this separate file? Can we just use the default Homebrew-installed Ruby file?

**Comment by staticfloat** [comment]
Ah, sorry about that.

The problem is this chunk:
```
pour_bottle? do
    reason "The bottle needs to be installed into /usr/local."
    # c.f. the identical hack in lua
    # https://github.com/Homebrew/homebrew/issues/47173
    satisfy { HOMEBREW_PREFIX.to_s == "/usr/local" }
  end
```

This stops our `Homebrew.jl` from using bottles, because it is not installed to `/usr/local`.  We don't want that.  We want to override this check.  So what we do is basically just copy the formula to our local tap, remove that chunk, force `Homebrew.jl` to use this formula file instead of the one from `Homebrew/core` and call it a day.

Can you verify that this formula on your machine works?

**Comment by staticfloat** [comment]
Oh, wait, it won't work as-is.  Try changing the bottle stanza URL to be the same as the mainline Homebrew one as well.  Let's just see if that works.

**Comment by NHDaly** [comment]
Got it. Thanks for the explanation!! :)

> Can you verify that this formula on your machine works?

:[ tbh i really don't understand how this all hooks together... How do I even pull this down to my machine? (I created this PR entirely through github...) Sorry! :) Thanks for your help if you don't mind.

> Try changing the bottle stanza URL to be the same as the mainline Homebrew one as well

I'll do that now.

**Comment by NHDaly** [comment]
> > Try changing the bottle stanza URL
> I'll do that now.

Haha jk, i actually don't know what you're referring to. Could you be more specific? Or, actually, I've checked the `allow edits from Maintainers` so you might be able to just change it right here on the PR page?

**Comment by staticfloat** [comment]
> :[ tbh i really don't understand how this all hooks together

No worries; after you install `Homebrew.jl`, this git repository gets downloaded to `~/.julia/v0.6/Homebrew/deps/usr/Library/Taps/staticfloat/homebrew-juliadeps`.  If you go to that directory, then check out this branch, you will then be able to `Homebrew.add("fontconfig")` or `Homebrew.add("cairo")` or what have you and see if these things get installed properly.

> i actually don't know what you're referring to.  Could you be more specific?

I'll make the change, then you'll see what I mean.  :)

**Comment by staticfloat** [comment]
Ah, no.  That doesn't work.  I forgot they embed URLs from the tap.  Grrrr.  Okay I guess I have to mirror it.  Gimme a sec.

**Comment by staticfloat** [comment]
Okay with the way things are now, it works on my machine.  You can doubly check by looking at `$(Pkg.dir("Cairo"))/deps/deps.jl`, which shows:

```
...
# Load dependencies
@checked_lib _jl_libgobject "/Users/sabae/.julia/v0.6/Homebrew/deps/usr/lib/libgobject-2.0.dylib"
@checked_lib _jl_libcairo "/Users/sabae/.julia/v0.6/Homebrew/deps/usr/lib/libcairo.dylib"
@checked_lib _jl_libpango "/Users/sabae/.julia/v0.6/Homebrew/deps/usr/lib/libpango-1.0.dylib"
@checked_lib _jl_libpangocairo "/Users/sabae/.julia/v0.6/Homebrew/deps/usr/lib/libpangocairo-1.0.dylib"
```

**Comment by NHDaly** [comment]
> this git repository gets downloaded to `~/.julia/v0.6/Homebrew/deps/usr/Library/Taps/staticfloat/homebrew-juliadeps`. If you go to that directory, then check out this branch,

Hmm for me, that directory isn't it's own repo. It's just part of the `~/.julia/v0.6/Homebrew` repo.  So I tried just removing the directory and cloning this repo in its place, which worked, but Homebrew still seems to be loading 2.12 somehow....

Anyway,  glad it's working for you now! let me know when this is live and i'll try Cairo again! :)

**Comment by NHDaly** [comment]
Oh, JK, I just realized after i cloned my Fork i forgot to checkout my branch.

Okay yeah, I replaced my clone with a clone of this repo at `master` now that it's merged, and yes, i got 2.13. :D Thanks!

**Comment by tknopp** [comment]
Thanks! Does something need to be released or should this already work? I just restarted a travis build on Gtk and this still fails with the wrong installation of fontconfig:

https://travis-ci.org/JuliaGraphics/Gtk.jl/jobs/415106130

**Comment by NHDaly** [comment]
(This was resolved here: https://github.com/JuliaGraphics/Cairo.jl/pull/243#issuecomment-413184631)

tl;dr It doesn't need to be released. `Homebrew.update()` will pull it down. Unclear why travis is still failing on 0.7.
